bindings/rust/s2n-tls-tokio/tests/shutdown.rs

```diff
@@ -281,3 +286,109 @@ async fn shutdown_with_poll_blinding() -> Result<(), Box<dyn std::error::Error>>
 
     Ok(())
 }
+
+#[tokio::test(start_paused = true)]
+async fn shutdown_with_tcp_error() -> Result<(), Box<dyn std::error::Error>> {
+    let client = TlsConnector::new(common::client_config()?.build()?);
+    let server = TlsAcceptor::new(common::server_config()?.build()?);
+
+    let (server_stream, client_stream) = common::get_streams().await?;
+    let server_stream = common::TestStream::new(server_stream);
+    let overrides = server_stream.overrides();
+
+    let (mut client, mut server) =
+        common::run_negotiate(&client, client_stream, &server, server_stream).await?;
+
+    // Attempt to shutdown the client. This will eventually fail because the
+    // server has not written the close_notify message yet, but it will at least
+    // write the close_notify message that the server needs.
+    // Because time is mocked for testing, this does not actually take 10 minutes.
+    // TODO: replace this with a half-close once the bindings support half-close.
+    _ = time::timeout(time::Duration::from_secs(600), client.shutdown()).await;
```

Comment:
Not necessarily in this PR, but I think we should add this as a suggestion in our documentation.
> Note that shutdown().await will wait for peer acknowledgement for an unbounded amount of time. Resource sensitive consumers should consider calling shutdown with an explicit timeout added

Or something like that.